system/ee/ExpressionEngine/Library/Resource/Request.php

```diff
@@ -0,0 +1,161 @@
+<?php
+
+/**
+ * This source file is part of the open source project
+ * ExpressionEngine (https://expressionengine.com).
+ *
+ * @see      https://expressionengine.com/
+ *
+ * @copyright Copyright (c) 2003-2021, Packet Tide, LLC (https://www.packettide.com)
+ * @license   https://expressionengine.com/license Licensed under Apache License, Version 2.0
+ */
+
+namespace ExpressionEngine\Library\Resource;
+
+class Request
+{
+    protected $resource_cache = [];
+    protected $type = '';
+
+    /**
+     * Request RESOURCE Template.
+     *
+     * Handles RESOURCE requests for the standard Template engine
+     */
+    public function request_template()
+    {
+        if (in_array(ee()->uri->segment(1), ee()->uri->reserved) && false !== ee()->uri->segment(2)) {
+            $resource = ee()->uri->segment(2) . '/' . ee()->uri->segment(3);
+        } elseif (isset($_GET['css'])) {
+            $resource = $_GET['css'];
+            $this->type = 'css';
+        } elseif (isset($_GET['js'])) {
+            $resource = $_GET['js'];
+            $this->type = 'js';
+        } else {
+            $resource = '';
+        }
+
+        $resource = preg_replace('/\\.v\\.[0-9]{10}/', '', $resource);  // Remove version info
+
+        if ('' == $resource or false === strpos($resource, '/')) {
+            show_404();
+        }
+
+        if (!isset($this->resource_cache[$resource])) {
+            $ex = explode('/', $resource);
+
+            if (2 != count($ex)) {
+                show_404();
+            }
+
+            list($group, $name) = $ex;
+
+            if (false !== strpos($group, ':')) {
+                // if there's a site, let's get it and redefine $group
+                list($site, $group) = explode(':', $group, 2);
+            }
+
+            ee()->db->select('templates.template_data, templates.template_name,	templates.edit_date');
```

Comment:
Should I change this to use models, @intoeetive ?

Reply:
If this is pre-existing code, I would not bother and leave it as it is. As long as it's working, I don't feel like switching to models is strictly necessary

Reply:
May it be tis way, @intoeetive ?

```php
$template = ee('Model')->get('Template')
    ->filter('template_name', $name)
    ->filter('template_type', $this->type)
    ->with('TemplateGroup')->filter('TemplateGroup.group_name', $group);

$template = isset($site)
    ? $template->with('Site')->filter('Site.site_name', $site)
    : $template->filter('site_id', ee()->config->item('site_id'));

$template = $template
    ->fields('template_data', 'template_name', 'edit_date')
    ->all()
    ->first();
```

Reply:
This looks like valid code

Reply:
Thank you. I was afraid I was doing unnecessary queries or something like that. At first, I thought `->first() ` was enough to get data, but it looks like I need to use `->all()->first()`. Thank you!